src/bootstrap/src/core/config/flags.rs

```diff
@@ -198,24 +198,34 @@ impl Flags {
             cmd: Kind,
         }
         if let Ok(HelpVerboseOnly { help: true, verbose: 1.., cmd: subcommand }) =
-            HelpVerboseOnly::try_parse_from(it.clone())
+            HelpVerboseOnly::try_parse_from(normalize_args(args))
         {
             println!("NOTE: updating submodules before printing available paths");
-            let config = Config::parse(&[String::from("build")]);
+            let config = Config::parse(Self::parse(&[String::from("build")]));
             let build = Build::new(config);
             let paths = Builder::get_help(&build, subcommand);
             if let Some(s) = paths {
                 println!("{s}");
             } else {
                 panic!("No paths available for subcommand `{}`", subcommand.as_str());
             }
-            crate::exit!(0);
+            true
+        } else {
+            false
         }
+    }
 
-        Flags::parse_from(it)
+    pub fn parse(args: &[String]) -> Self {
+        Flags::parse_from(normalize_args(args))
     }
```

Comment:
Would be nice to cover `normalize_args` behavior with a unit test for the parse function.

Reply:
I'm not sure if I understand what exactly do you want to test. Just the fact that `normalize_args` adds `x.py` at the beginnin?

Reply:
I misunderstood the logic there. Why are we adding 'x.py' in the first place? If we want the first argument from the invoked command, we could remove the 'skip' part from [here](https://github.com/rust-lang/rust/blob/c9bd03cb724e13cca96ad320733046cbdb16fbbe/src/bootstrap/src/bin/main.rs#L19).

Reply:
Again, no idea :sweat_smile: This is what bootstrap did before, I just moved stuff around. I suppose that it was done to normalize `x`, `x.py` etc., but I don't know why exactly it needs to be normalized. This was implemented back in 2015, in the first commit that created the Rust version of bootstrap :laughing:

Reply:
I think it's done simply because a lot of code in bootstrap (mostly in tests) invokes `Config::parse` with an array of strings, and it doesn't bother to add the "command name" to that array.

Reply:
Yeah, it is because of that. We could clean-up this mess in another PR, feel free to r=me.